tests/evals/general_correctness_check.py

```diff
@@ -0,0 +1,1286 @@
+
+    # Run evaluation
+    python tests/evals/general_correctness_check.py \
+        --base-url http://localhost:8000 \
+        --model meta-llama/Llama-3.1-8B
+
+    # Quick test with fewer samples
+    python tests/evals/general_correctness_check.py \
+        --base-url http://localhost:8000 \
+        --model meta-llama/Llama-3.1-8B \
+        --gsm8k-samples 20 \
+        --sampling-prompts 10
+"""
+
+import argparse
+import asyncio
+import json
+import sys
+import time
+from pathlib import Path
+from typing import Any
+
+import aiohttp
+import numpy as np
+from tqdm.asyncio import tqdm
+
+# Add parent directory to path for imports
+sys.path.insert(0, str(Path(__file__).parent.parent.parent))
+
+# Import GSM8K utilities
+from tests.evals.gsm8k.gsm8k_eval import (
+    INVALID,
+    call_vllm_api,
+    get_answer_value,
+    load_gsm8k_data,
+)
+
+
+class GeneralCorrectnessChecker:
+    """General correctness checker for vLLM endpoints."""
+
+    def __init__(self, base_url: str, model_name: str | None = None):
+        self.base_url = base_url.rstrip("/")
+        self.model_name = model_name
+        self._model_detected = False
+
+    async def detect_model(self) -> str | None:
+        """
+        Auto-detect the model being served by querying /v1/models.
+
+        Returns:
+            Model name if detected, None otherwise
+        """
+        try:
+            async with (
+                aiohttp.ClientSession() as session,
+                session.get(
+                    f"{self.base_url}/v1/models",
+                    timeout=aiohttp.ClientTimeout(total=10),
+                ) as response,
+            ):
+                if response.status == 200:
+                    data = await response.json()
+                    models = data.get("data", [])
+                    if models and len(models) > 0:
+                        model_name = models[0].get("id")
+                        self._model_detected = True
+                        return model_name
+        except Exception as e:
+            print(f"Could not auto-detect model: {e}")
+        return None
+
+    async def check_health(self) -> bool:
+        """Check if the endpoint is healthy."""
+        try:
+            # Health endpoint is at /health, not /v1/health
+            # Remove /v1 suffix if present
+            health_url = self.base_url.rstrip("/")
+            if health_url.endswith("/v1"):
+                health_url = health_url[:-3]
+            health_url = f"{health_url}/health"
+
+            async with (
+                aiohttp.ClientSession() as session,
+                session.get(
+                    health_url, timeout=aiohttp.ClientTimeout(total=10)
+                ) as response,
+            ):
+                return response.status == 200
+        except Exception as e:
+            print(f"Health check failed: {e}")
+            return False
+
+    async def evaluate_gsm8k_correctness(
```

Comment:
It looks to me like `evaluate_gsm8k()` is already a method in `gsm8k_eval` that does most of this